fizzbuzz.js

```diff
@@ -0,0 +1,14 @@
+const fizzbuzz = () => {
+    for (let i=1; i<=100; i++) {
+        if (i%3===0 && i%5===0) {
+            console.log("FizzBuzz")
+        }
+        else if (i%3===0){
+            console.log("Fizz")
+        }
+        else if (i%5 === 0) {
+            console.log("Buzz")
+        }
+        else console.log(i)
+    }
+}
```

Comment:
Instead of imperatively looping `n` times, you could create a `range` helper instead
```javascript
const range = n => new Array(n).fill().map((_, i) => i)
``` 
In the first exercise, you are mutating board variable, it's fine. But such patterns are really nasty and hard to reason about in the production code. Try to express it without any mutations. For example, it could look like this for the fizzBuzz.
```javascript
const range = n => new Array(n).fill().map((_, i) => i)

const fizz = n => s => n%3===0 ? s + 'Fizz' : s
const buzz = n => s => n%5===0 ? s + 'Buzz' : s
const fizzBuzz = i => compose(buzz(i),fizz(i))('') || i

const run = n => console.log(range(n).map(fizzBuzz).join('\n'))
```

Reply:
Well, I don't agree with these remarks @grzegorz-bielski .

I like the idea of `range` helper, but should we demand to omit using in example `for` loops in exercise to the chapter, where a student is learning about... loops? I know that we prefer other solutions in real projects, but do we want someone to slight the examples from the book they are learning from? I'm not sure :|

Also, I really don't like the second snippet. I'm not a fan of providing complicated solutions for trivial problems. In that case, I even prefer loops way of solving that problem then that complicated set of composition and higher-order functions. Of course, we can provide some cleaner solutions with functions like map or reduce, but I was writing about that in the paragraph above.

Let me know what you think ;)

Reply:
Yea, you are right. I kind of jumped ahead too far with this second snippet, but I didn't ask anyone to implement this that way though, nor I think that my solution is best out there. That's just an example of a different kind of approach, an exercise in itself. Besides, a programmer doesn't need to know right away of inner workings of some tools to use them, sometimes an example, like this one for `range` is just enough.

I wouldn't necessarily agree that loops are simpler than function composition or higher-order functions, it's just that JS syntax makes it harder than necessary and everyone is learning loops with mutable variables as basic concepts and unlearning them leveling up... It doesn't need to be always that way (In fact, [it isn't)](https://web.mit.edu/alexmv/6.037/sicp.pdf), and there are significant benefits [to this approach.](https://www.youtube.com/watch?v=bmFKEewRRQg&t=1979s) 

Anyway, right... let's stick (mostly) to the book for the time being. 🐤